crates/ethereum-rpc/src/lib.rs

```diff
@@ -51,503 +53,613 @@ pub struct SyncStatus {
+            .expect("Tx index must be set for tx inside block");
+
+        let block_number: u64 = tx
+            .block_number
+            .expect("Block number must be set for tx inside block");
+
+        let traces = debug_trace_by_block_number(
+            block_number,
+            Some(trace_idx as usize),
+            &self.ethereum,
+            &evm,
+            &mut working_set,
+            opts,
+        )
+        .map_err(to_eth_rpc_error)?;
+
+        Ok(traces[0].clone())
+    }
 
-    rpc.register_async_method("txpool_content", |_, _, _| async move {
+    fn txpool_content(&self) -> RpcResult<Value> {
         // This is a simple mock for serde.
-        let json = json!({
+        Ok(json!({
             "pending": {},
             "queued": {}
-        });
-
-        Ok::<_, ErrorObjectOwned>(json)
-    })?;
-
-    rpc.register_async_method(
-        "eth_getUncleByBlockHashAndIndex",
-        |parameters, _, _| async move {
-            let mut params = parameters.sequence();
-
-            let _block_hash: String = params.next()?;
-            let _uncle_index_position: String = params.next()?;
-
-            let res = json!(null);
-
-            Ok::<_, ErrorObjectOwned>(res)
-        },
-    )?;
-
-    if !is_sequencer {
-        rpc.register_async_method::<Result<B256, ErrorObjectOwned>, _, _>(
-            "eth_sendRawTransaction",
-            |parameters, ethereum, _| async move {
-                // send this directly to the sequencer
-                let data: Bytes = parameters.one()?;
-                // sequencer client should send it
-                let tx_hash = ethereum
-                    .sequencer_client
-                    .as_ref()
-                    .unwrap()
-                    .eth_send_raw_transaction(data)
-                    .await;
+        }))
+    }
 
-                match tx_hash {
-                    Ok(tx_hash) => Ok(tx_hash),
+    fn get_uncle_by_block_hash_and_index(
+        &self,
+        _block_hash: String,
+        _uncle_index: String,
+    ) -> RpcResult<Value> {
+        Ok(json!(null))
+    }
+
+    async fn eth_send_raw_transaction(&self, data: Bytes) -> RpcResult<B256> {
+        let tx_hash = SequencerRpcClient::eth_send_raw_transaction(
+            self.ethereum.sequencer_client.as_ref().unwrap(),
+            data,
+        )
+        .await;
+
+        match tx_hash {
+            Ok(tx_hash) => Ok(tx_hash),
+            Err(e) => match e {
+                jsonrpsee::core::client::Error::Call(e_owned) => Err(e_owned),
+                _ => Err(to_jsonrpsee_error_object("SEQUENCER_CLIENT_ERROR", e)),
+            },
+        }
+    }
+
+    async fn eth_get_transaction_by_hash(
+        &self,
+        hash: B256,
+        mempool_only: Option<bool>,
+    ) -> RpcResult<Option<RpcTransaction<AnyNetwork>>> {
+        match mempool_only {
+            Some(true) => {
+                match SequencerRpcClient::eth_get_transaction_by_hash(
```

Comment:
why not self.ethereum.sequencer_client.as_ref().unwrap().eth_get_transaction_by_hash() ?